config/gh-actions/deploy-surge.sh

```diff
@@ -52,16 +63,40 @@ do
     continue
   fi
 
-  DASHBOARD_DEPLOY_DOMAIN=https://dashboard-${DEPLOY_SUBDOMAIN}-${REPO_NAME}-${REPO_OWNER}.surge.sh
-  echo "Deploy domain: ${DASHBOARD_DEPLOY_DOMAIN}"
-  surge --project ${DASHBOARD_DEPLOY_PATH} --domain $DASHBOARD_DEPLOY_DOMAIN;
+  if [ "$ACTIONS_STATUS" == "failure" ]
+  then
+    # Update GitHub status to failed
+    curl "https://api.github.com/repos/${REPO_OWNER}/${REPO_NAME}/statuses/${ACTIONS_PULL_REQUEST_HEAD}?access_token=${GITHUB_TOKEN}" \
+    -H "Content-Type: application/json" \
+    -X POST \
+    -d "{\"state\": \"failure\",\"context\": \"dashboard/surge/deploy/${DEPLOY_SUBDOMAIN}\", \"description\": \"Dashboard deploy failed\", \"target_url\": \"${ACTIONS_WORKFLOW_RUN_URL}\"}"
 
-  MARKBIND_DEPLOY_DOMAIN=https://docs-${DEPLOY_SUBDOMAIN}-${REPO_NAME}-${REPO_OWNER}.surge.sh
-  echo "Deploy domain: ${MARKBIND_DEPLOY_DOMAIN}"
-  surge --project ${MARKBIND_DEPLOY_PATH} --domain $MARKBIND_DEPLOY_DOMAIN;
+    curl "https://api.github.com/repos/${REPO_OWNER}/${REPO_NAME}/statuses/${ACTIONS_PULL_REQUEST_HEAD}?access_token=${GITHUB_TOKEN}" \
```

Comment:
To further reduce the length of the shell script, can you look into shell functions to abstract the curl method?

Reply:
The variable will be state, description, and target url?